tests/unit_tests/ray_tracing/test_mirror_panel_psf.py

```diff
@@ -0,0 +1,357 @@
+        "test": False,
+        "psf_measurement": "tests/resources/MLTdata-preproduction.ecsv",
+        "psf_measurement_containment_mean": 0.5,
+        "psf_measurement_containment_sigma": 0.1,
+        "site": "North",
+        "telescope": "LSTN-01",
+        "model_version": "test_version",
+        "mirror_list": None,
+        "random_focal_length": None,
+        "no_tuning": False,
+        "rnda": 0,
+        "simtel_path": "path/to/simtel",
+        "number_of_mirrors_to_test": 2,
+        "use_random_focal_length": False,
+        "containment_fraction": 0.8,
+        "output_path": "",
+    }
+
+    with (
+        patch(mock_telescope_model_string),
+        patch(mock_find_file_string),
+        patch("simtools.ray_tracing.mirror_panel_psf.RayTracing") as mock_ray_tracing,
+    ):
+
+        db_config = {"db": "config"}
+        label = "test_label"
+        mirror_panel_psf = MirrorPanelPSF(label, args_dict, db_config)
+
+        mock_ray_instance = mock_ray_tracing.return_value
+        mock_ray_instance.get_mean.return_value = 0.5 * u.cm
+        mock_ray_instance.get_std_dev.return_value = 0.1 * u.cm
+
+        mean_d80, sig_d80 = mirror_panel_psf.run_simulations_and_analysis(rnda)
+
+        mirror_panel_psf.telescope_model.change_parameter.assert_called_once_with(
+            "mirror_reflection_random_angle", rnda
+        )
+        mock_ray_tracing.assert_called_once_with(
+            telescope_model=mirror_panel_psf.telescope_model,
+            simtel_path=mirror_panel_psf.args_dict.get("simtel_path", None),
+            single_mirror_mode=True,
+            mirror_numbers=(
+                list(range(1, mirror_panel_psf.args_dict["number_of_mirrors_to_test"] + 1))
+                if mirror_panel_psf.args_dict["test"]
+                else "all"
+            ),
+            use_random_focal_length=mirror_panel_psf.args_dict["use_random_focal_length"],
+        )
+        mock_ray_instance.simulate.assert_called_once_with(
+            test=mirror_panel_psf.args_dict["test"], force=True
+        )
+        mock_ray_instance.analyze.assert_called_once_with(force=True)
+        assert mean_d80 == 0.5
+        assert sig_d80 == 0.1
+
+
+def test_print_results(mock_mirror_panel_psf, capsys):
+    mirror_panel_psf = copy.deepcopy(mock_mirror_panel_psf)
+    mirror_panel_psf.results_rnda = [0.1, 0.2, 0.3]
+    mirror_panel_psf.results_mean = [0.4, 0.5, 0.6]
+    mirror_panel_psf.results_sig = [0.01, 0.02, 0.03]
+    mirror_panel_psf.rnda_opt = 0.25
+    mirror_panel_psf.rnda_start = 0.3
+    mirror_panel_psf.mean_d80 = 0.55
+    mirror_panel_psf.sig_d80 = 0.025
+
+    mirror_panel_psf.print_results()
+    out = capsys.readouterr().out
+    assert "StdDev" in out
+    assert "New value" in out
+
+    mirror_panel_psf.args_dict["psf_measurement_containment_sigma"] = None
+    mirror_panel_psf.print_results()
+    out = capsys.readouterr().out
+    assert "StdDev = 0.010 cm" not in out
+    assert "New value" in out
```

Comment:
Not fully sure about this one here. We use the same out as above without deleting in between and check for the same string. Probably need to be cleared in between?

Reply:
Surprisingly little in the documentation for capsys, but found this in the source code:

```
def readouterr(self) -> CaptureResult[AnyStr]:
        """Read and return the captured output so far, resetting the internal
        buffer.
```

So the buffers are clearer.

I also checked this with simple print statements for above test function and confirm that buffers are cleared.

Reply:
Okey good, thanks for checking!

Reply:
And you are free to merge